kernel/tests/write.rs

```diff
@@ -251,3 +270,382 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
+            assert!((now - commit_info_ts.as_i64().unwrap()).abs() < 10_000);
+        }
+        if let Some(add_ts) = &commit.pointer("/add/modificationTime") {
+            assert!((now - add_ts.as_i64().unwrap()).abs() < 10_000);
+        }
+    });
+
+    Ok(())
+}
+
+// update `value` at (.-separated) `path` to `new_value`
+fn set_value(
+    value: &mut serde_json::Value,
+    path: &str,
+    new_value: serde_json::Value,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let mut path_string = path.replace(".", "/");
+    path_string.insert(0, '/');
+    let v = value
+        .pointer_mut(&path_string)
+        .ok_or_else(|| format!("key '{path}' not found"))?;
+    *v = new_value;
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_append() -> Result<(), Box<dyn std::error::Error>> {
+    // setup tracing
+    let _ = tracing_subscriber::fmt::try_init();
+    // setup in-memory object store and default engine
+    let (store, engine, table_location) = setup("test_table", true);
+
+    // create a simple table: one int column named 'number'
+    let schema = Arc::new(StructType::new(vec![StructField::new(
+        "number",
+        DataType::INTEGER,
+        true,
+    )]));
+    let table = create_table(store.clone(), table_location, schema.clone(), &[]).await?;
+
+    let commit_info = new_commit_info()?;
+
+    let mut txn = table
+        .new_transaction(&engine)?
+        .with_commit_info(commit_info);
+
+    // create two new arrow record batches to append
+    let append_data = [[1, 2, 3], [4, 5, 6]].map(|data| -> DeltaResult<_> {
+        let data = RecordBatch::try_new(
+            Arc::new(schema.as_ref().try_into()?),
+            vec![Arc::new(arrow::array::Int32Array::from(data.to_vec()))],
+        )?;
+        Ok(Box::new(ArrowEngineData::new(data)))
+    });
+
+    // write data out by spawning async tasks to simulate executors
+    let engine = Arc::new(engine);
+    let write_context = Arc::new(txn.get_write_context());
+    let tasks = append_data.into_iter().map(|data| {
+        // arc clones
+        let engine = engine.clone();
+        let write_context = write_context.clone();
+        tokio::task::spawn(async move {
+            engine
+                .write_parquet(
+                    data.as_ref().unwrap(),
+                    write_context.as_ref(),
+                    HashMap::new(),
+                    true,
+                )
+                .await
+        })
+    });
+
+    let write_metadata = futures::future::join_all(tasks).await.into_iter().flatten();
+    for meta in write_metadata {
+        txn.add_write_metadata(meta?);
+    }
+
+    // commit!
+    txn.commit(engine.as_ref())?;
+
+    let commit1 = store
+        .get(&Path::from(
+            "/test_table/_delta_log/00000000000000000001.json",
+        ))
+        .await?;
+
+    let mut parsed_commits: Vec<_> = Deserializer::from_slice(&commit1.bytes().await?)
+        .into_iter::<serde_json::Value>()
+        .try_collect()?;
```

Comment:
nit: Does this work, out of curiosity?
```suggestion
    let commit1 = commit1.bytes().await?;
    let mut parsed_commits: Vec<serde_json::Value> = Deserializer::from_slice(&commit1)
        .into_iter()
        .try_collect()?;
```
(but actually it's harder to read that than the turbofish, so meh)

Reply:
it does work! but yea i agree original is more readable to me?